include/envoy/stream_info/stream_info.h

```diff
@@ -236,6 +236,11 @@ class StreamInfo {
    */
   virtual void setResponseFlag(ResponseFlag response_flag) PURE;
 
+  /**
+   * @param code the HTTP response code to set for this request.
+   */
+  virtual void setResponseCode(uint32_t code) PURE;
```

Comment:
The callers of this function pass in Http::Code but the underlying type is uint32_t, so I went with that.

Open to suggestions as to whether we think this should remain uint32_t or change to Http::Code.